### PR TITLE
Create protocol details from the app-mgt component

### PR DESCRIPTION
**Pull request by sahandilshan**
### Proposed changes in this pull request
- $subject
- Creates the protocol details along with the service provider

With this PR, we will be able to handle the inbound protocol while handling (while doing CRUD operations) the application management. Earlier we had to separately call app-mgt services and protocol services to do this. With this improvement, the app-mgt service will be able to handle the protocol configurations as well.

image

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java [hunk 1]
```diff
@@ -174,4 +191,52 @@ private ApplicationInboundTableColumns() {
...
+
+    /**
+     * Enums for error messages.
+     */
+    public enum ErrorMessage {
+
+        ERROR_RETRIEVING_USER_BY_ID("65503", "Error occurred while retrieving user",
+                "Error occurred while retrieving user by userid: %s."),
+        NON_EXISTING_USER_ID("60504", "User not found",
+                "Non exiting user for the given userid: %s."),
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java:203]:
```suggestion
                "No user found for the given user-id: %s."),
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java [hunk 2]
```diff
@@ -174,4 +191,52 @@ private ApplicationInboundTableColumns() {
...
+        ERROR_RETRIEVING_USERSTORE_MANAGER("65504", "Error retrieving userstore manager.",
+                "Error occurred while retrieving userstore manager."),
+        UNEXPECTED_ERROR("65006", "Unexpected Processing Error.",
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java:206]:
```suggestion
        UNEXPECTED_ERROR("65006", "Unexpected processing error.",
```

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/IdentityApplicationManagementClientException.java [hunk 3]
```diff
@@ -90,6 +90,31 @@ public IdentityApplicationManagementClientException(String errorCode, String mes
         super(errorCode, message, cause);
     }
 
+    /**
+     * Constructs a new exception with the specified error code, message and description.
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/IdentityApplicationManagementClientException.java:94]: Missing new line after the description

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/IdentityApplicationManagementClientException.java [hunk 4]
```diff
@@ -90,6 +90,31 @@ public IdentityApplicationManagementClientException(String errorCode, String mes
...
+     * @param errorCode Error code.
+     * @param message Error message.
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/IdentityApplicationManagementClientException.java:96]: Param descriptions should be vertically aligned

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 5]
```diff
@@ -2522,6 +2537,84 @@ public String createApplication(ServiceProvider application, String tenantDomain
...
+                    application, applicationModel.getInboundProtocolConfigurationDto());
+            
+            return createApplication(application, tenantDomain, username);
+        } catch (Throwable e) {
+            /*
+             * For more information read https://github.com/wso2/product-is/issues/12579. This is to overcome the
+             * above issue.
+             */
+            if (log.isDebugEnabled()) {
+                log.debug("Server encountered unexpected error. Rolling back created application data.", e);
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2566]: We don't need the `isDebugEnabled` check

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 6]
```diff
@@ -2522,6 +2537,84 @@ public String createApplication(ServiceProvider application, String tenantDomain
...
+            ServiceProvider application, InboundProtocolsDTO inboundProtocolsModel)
+            throws IdentityApplicationManagementException {
+
+        if (inboundProtocolsModel == null) {
+            return null;
+        }
+        List<InboundAuthenticationRequestConfig> addedInbounds = new ArrayList<>();
+        try {
+            // This handles the inbound protocol creation with the help of protocol handlers.
+            List<ApplicationInboundAuthConfigHandler> applicationInboundAuthConfigHandlerList =
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2584]: `applicationInboundAuthConfigHandlerList` can this get null?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/ApplicationManagementServiceComponentHolder.java [hunk 7]
```diff
@@ -382,4 +387,32 @@ public void setSecretResolveManager(SecretResolveManager secretResolveManager) {
...
+     * @param applicationInboundAuthConfigHandler Protocol service.
+     */
+    public void removeApplicationInboundConfigHandler(ApplicationInboundAuthConfigHandler
+                                                              applicationInboundAuthConfigHandler) {
+        
+        applicationInboundAuthConfigHandlers.remove(applicationInboundAuthConfigHandler);
+    }
+
+    /**
+     * Get application inbound config service.
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/ApplicationManagementServiceComponentHolder.java:411]: New line missing

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/ApplicationManagementServiceComponentHolder.java [hunk 8]
```diff
@@ -382,4 +387,32 @@ public void setSecretResolveManager(SecretResolveManager secretResolveManager) {
         this.secretResolveManager = secretResolveManager;
     }
 
+
+    /**
+     * Add application inbound config service.
+     */
+    public void addApplicationInboundAuthConfigHandler(ApplicationInboundAuthConfigHandler
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/ApplicationManagementServiceComponentHolder.java:394]: Don't we need to import this package in the pom.xml?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/protocol/ProtocolUtils.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+/**
+ * Utility functions related to application inbound protocols.
+ */
+public class ProtocolUtils {
+
+    public static String getInboundAuthKey(ServiceProvider application,
+                                           String inboundType) {
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/protocol/ProtocolUtils.java:34]:
```suggestion
    public static String getInboundAuthKey(ServiceProvider application, String inboundType) {
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java (hunk 1 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java]: Fixed with Fixed with e4bf79511c2924cb02a2721d4e99f48a0761156f

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java (hunk 2 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java]: Fixed with e4bf79511c2924cb02a2721d4e99f48a0761156f

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/IdentityApplicationManagementClientException.java (hunk 4 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/IdentityApplicationManagementClientException.java]: Fixed with e4bf79511c2924cb02a2721d4e99f48a0761156f

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 10]
```diff
@@ -2624,15 +2718,126 @@ public void updateApplicationByResourceId(String resourceId, ServiceProvider upd
...
-        if (ApplicationMgtUtil.isLegacyAuditLogsDisabledInAppMgt()) {
+        if (ApplicationMgtUtil.isEnableV2AuditLogs()) {
             AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(
                     getInitiatorId(username, tenantDomain), USER, resourceId, TARGET_APPLICATION,
                     ApplicationConstants.LogConstants.UPDATE_APPLICATION)
                     .data(buildSPData(updatedApp));
             triggerAuditLogEvent(auditLogBuilder, true);
         }
     }
+
```
[the remark this answers is not in the document]
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2726]: Fixed with 8f83c6b4be41c5708dd29fbb642aa757c2360109

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/ApplicationManagementServiceComponentHolder.java (hunk 7 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/ApplicationManagementServiceComponentHolder.java:411]: Fixed with ac274f604574892ddb791b5024e3d57bc51817fa

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 11]
```diff
@@ -2522,6 +2536,84 @@ public String createApplication(ServiceProvider application, String tenantDomain
...
+            throws IdentityApplicationManagementException {
+
+        List<InboundAuthenticationRequestConfig> addedInbounds = new ArrayList<>();
+        try {
+            ServiceProvider application = applicationModel.getServiceProvider();
+             addedInbounds = addInboundAuthenticationProtocolsToApplication(
+                    application, applicationModel.getInboundProtocolConfigurationDto());
+            
+            return createApplication(application, tenantDomain, username);
+        } catch (Throwable e) {
```
Comment (sahandilshan) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2559]: We should not catch Throwable

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 12]
```diff
@@ -2522,6 +2536,84 @@ public String createApplication(ServiceProvider application, String tenantDomain
...
+            /*
+             * For more information read https://github.com/wso2/product-is/issues/12579. This is to overcome the
```
Comment (sahandilshan) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2561]: Add more descriptive comment.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 13]
```diff
@@ -2522,6 +2536,84 @@ public String createApplication(ServiceProvider application, String tenantDomain
...
+                    "creating the application.", e);
+        }
+    }
+
+    private List<InboundAuthenticationRequestConfig> addInboundAuthenticationProtocolsToApplication(
+            ServiceProvider application, InboundProtocolsDTO inboundProtocolsModel)
+            throws IdentityApplicationManagementException {
+
+        if (inboundProtocolsModel == null) {
+            return null;
```
Comment (sahandilshan) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2578]: Do not return null. Return an empty list

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java [hunk 14]
```diff
@@ -0,0 +1,165 @@
...
+     */
+    public static void rollbackInbound(InboundAuthenticationRequestConfig inbound)
+            throws IdentityApplicationManagementException {
+        
+        List<ApplicationInboundAuthConfigHandler> applicationInboundAuthConfigHandlerList =
+                ApplicationManagementServiceComponentHolder.getInstance().getApplicationInboundAuthConfigHandler();
+        for (ApplicationInboundAuthConfigHandler applicationInboundAuthConfigHandler :
+                applicationInboundAuthConfigHandlerList) {
+            if (applicationInboundAuthConfigHandler.canHandle(inbound.getInboundAuthType())) {
+                applicationInboundAuthConfigHandler.handleConfigDeletion(inbound.getInboundAuthKey());
```
Comment (shanggeeth) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java:101]: Can we add a break here?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/dto/ApplicationDTO.java [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+
+        return serviceProvider;
+    }
+
+    public InboundProtocolsDTO getInboundProtocolConfigurationDto() {
+
+        return inboundProtocolConfigurationDto;
+    }
+
+
```
Comment (shanggeeth) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/dto/ApplicationDTO.java:78]: Remove this extra newline

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java (hunk 5 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: Fixed with 51b85af647109d22ee800443e12e61cc0b09c1dc

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/ApplicationManagementServiceComponentHolder.java [hunk 16]
```diff
@@ -382,4 +387,32 @@ public void setSecretResolveManager(SecretResolveManager secretResolveManager) {
...
+                                                               applicationInboundAuthConfigHandler) {
+        
+        applicationInboundAuthConfigHandlers.add(applicationInboundAuthConfigHandler);
+    }
+
+    /**
+     * Remove application inbound config service.
```
[the remark this answers is not in the document]
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/ApplicationManagementServiceComponentHolder.java:402]: Fixed with 51b85af647109d22ee800443e12e61cc0b09c1dc

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/protocol/ProtocolUtils.java (hunk 9 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/protocol/ProtocolUtils.java]: Fixed with 51b85af647109d22ee800443e12e61cc0b09c1dc

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java (hunk 14 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java:101]: Fixed with 51b85af647109d22ee800443e12e61cc0b09c1dc

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/dto/ApplicationDTO.java (hunk 15 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/dto/ApplicationDTO.java]: Fixed with 51b85af647109d22ee800443e12e61cc0b09c1dc

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+            InboundAuthenticationRequestConfig[] inbounds = inboundAuthConfig.getInboundAuthenticationRequestConfigs();
+            if (inbounds != null) {
+                return Arrays.stream(inbounds)
+                        .filter(inbound -> inboundType.equals(inbound.getInboundAuthType()))
+                        .findAny()
+                        .map(InboundAuthenticationRequestConfig::getInboundAuthKey)
+                        .orElse(null);
+            }
+        }
+        return null;
```
Comment (sahandilshan) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java:70]: Add optional here

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 18]
```diff
@@ -2522,6 +2536,83 @@ public String createApplication(ServiceProvider application, String tenantDomain
...
+                    ApplicationManagementServiceComponentHolder.getInstance().getApplicationInboundAuthConfigHandler();
+            for (ApplicationInboundAuthConfigHandler applicationInboundAuthConfigHandler :
+                    applicationInboundAuthConfigHandlerList) {
+                if (applicationInboundAuthConfigHandler.canHandle(inboundProtocolsModel)) {
+                    addedInbounds.add(applicationInboundAuthConfigHandler.handleConfigCreation(application,
+                            inboundProtocolsModel));
+                }
+            }
+        } catch (IdentityApplicationManagementClientException e) {
+            // This is to void converting client exception in to a server exception.
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2592]: Shall we add a comment on why roll back is not required for the client exceptions

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 19]
```diff
@@ -2624,15 +2714,64 @@ public void updateApplicationByResourceId(String resourceId, ServiceProvider upd
...
+                break;
+            }
+        }
+        if (addedInbound != null) {
+            updateOrInsertInbound(serviceProvider, addedInbound);
+        }
+        try {
+            // At this point the serviceProvided object is updated with the inbound auth config details. So we can call
+            // the updateApplicationByResourceId(String, ServiceProvider, String, String) method to update the
+            // service provider details in the database.
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2766]: Use block comments

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 20]
```diff
@@ -2624,15 +2714,64 @@ public void updateApplicationByResourceId(String resourceId, ServiceProvider upd
...
+            if (log.isDebugEnabled()) {
+                log.debug("Inbound protocol configuration is not provided for the application: " + resourceId +
+                        ". Hence updating the application just using the service provider details in the" +
+                        " serviceProvider object.");
+            }
+            updateApplicationByResourceId(resourceId, serviceProvider, tenantDomain, username);
+            return;
+        }
+        InboundAuthenticationRequestConfig addedInbound = null;
+        List<ApplicationInboundAuthConfigHandler> applicationInboundAuthConfigHandler =
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2751]:
```suggestion
        List<ApplicationInboundAuthConfigHandler> applicationInboundAuthConfigHandlers =
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 21]
```diff
@@ -987,6 +994,33 @@ private static void maskClientSecret(JSONObject inboundAuthenticationConfig) {
             }
         }
     }
+    
+    /**
+     * Check whether the v2 audit logs are enabled.
+     * @return true if v2 audit logs are enabled.
+     */
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:1001]:
```suggestion
    /**
     * Check whether the v2 audit logs are enabled.
     *
     * @return true if v2 audit logs are enabled.
     */
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java [hunk 22]
```diff
@@ -0,0 +1,166 @@
...
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Utility functions related to application inbound protocols.
+ */
+public class InboundFunctions {
+    
+    private static final Log log = LogFactory.getLog(InboundFunctions.class);
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java:43]:
```suggestion
    private static final Log LOG = LogFactory.getLog(InboundFunctions.class);
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java [hunk 23]
```diff
@@ -0,0 +1,166 @@
...
+        for (ApplicationInboundAuthConfigHandler applicationInboundAuthConfigHandler :
+                applicationInboundAuthConfigHandlerList) {
+            if (applicationInboundAuthConfigHandler.canHandle(inbound.getInboundAuthType())) {
+                applicationInboundAuthConfigHandler.handleConfigDeletion(inbound.getInboundAuthKey());
+                break;
+            }
+        }
+    }
+    
+    public static void doRollback(String applicationId, InboundAuthenticationRequestConfig updatedInbound,
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java:107]: Add method description

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java (hunk 17 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java]: Fixed with 2eb7f05bc4f9dd13bebb13286d5b205745ffdad9

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java (hunk 19 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: Fixed with 2eb7f05bc4f9dd13bebb13286d5b205745ffdad9

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java (hunk 21 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java]: Fixed with 2eb7f05bc4f9dd13bebb13286d5b205745ffdad9

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java (hunk 22 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java]: Fixed with 2eb7f05bc4f9dd13bebb13286d5b205745ffdad9

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java (hunk 23 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java]: Fixed with 2eb7f05bc4f9dd13bebb13286d5b205745ffdad9

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java (hunk 18 above)
Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: Fixed with 2eb7f05bc4f9dd13bebb13286d5b205745ffdad9

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationResourceManager.java [hunk 24]
```diff
@@ -70,6 +85,14 @@ ServiceProvider getApplicationByResourceId(String resourceId,
     void updateApplicationByResourceId(String resourceId, ServiceProvider updatedApplication,
                                        String tenantDomain, String username)
             throws IdentityApplicationManagementException;
+    
+    default void updateApplicationByResourceId(String resourceId, ServiceProvider application,
```
Comment (pamodaaw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationResourceManager.java:89]: Add the method description

Comment (sahandilshan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationResourceManager.java:99]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/5146/commits/52ad1c5d357df775755cc1d1f76c7a76c4dd50ae

**Comment by sahandilshan** [comment]
The last commit https://github.com/wso2/carbon-identity-framework/pull/5146/commits/52ad1c5d357df775755cc1d1f76c7a76c4dd50ae only update some comments in the code, hence no need to re-run the PR builder again

**Review by pamodaaw — APPROVED** [review]

**Pull request merged**